src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -260,7 +260,12 @@ public OptionSet Options
 
         internal void UpdateCurrentSolutionOnOptionsChanged()
         {
-            SetCurrentSolution(CurrentSolution.WithOptions(new SolutionOptionSet(_legacyOptions)));
+            using (_serializationLock.DisposableWait())
+            {
+                SetCurrentSolution(CurrentSolution.WithOptions(new SolutionOptionSet(_legacyOptions)));
+
+                // TODO: Are we missing a call to RaiseWorkspaceChangedEventAsync here?
+            }
```

Comment:
two things of concern:

1. we werene't locking consistently when calling SetCurrentSolution (like everywhere else in this type)
2. we're not issueing an event here.